spot-client/src/spot-tv/app-state/actions.js

```diff
@@ -325,18 +330,15 @@ export function createSpotTVRemoteControlConnection({ pairingCode, retry }) {
  * to be consumed by a Spot-TV client.
  *
  * @param {Object} state - The Redux state.
+ * @param {SpotTvBackendService} [backend] - The backend instance that will be used with the RCS service.
  * @param {string} [permanentPairingCode] - This one is optional and used only in the backend scenario where Spot TV
  * needs a permanent pairing code in order to connect to the service.
  * @returns {Promise<Object>} Resolves with an object containing the latest
  * join code and jwt.
  */
-function createConnection(state, permanentPairingCode) {
+function createConnection(state, backend, permanentPairingCode) {
     const joinCodeRefreshRate = getJoinCodeRefreshRate(state);
     const remoteControlConfiguration = getRemoteControlServerConfig(state);
-    const backend
```

Comment:
Why pass it in? I'm not against the change but wonder why you chose to do it.

Reply:
I've tried to explain it in the commit message. The scope change is needed to preserve the exit password cache when internet is lost and the reconnect is in progress.